go/inst/instance_topology.go

```diff
@@ -2179,9 +2179,12 @@ func chooseCandidateReplica(replicas [](*Instance)) (candidateReplica *Instance,
 	replicas = RemoveInstance(replicas, &candidateReplica.Key)
 	for _, replica := range replicas {
 		replica := replica
-		if canReplicate, _ := replica.CanReplicateFrom(candidateReplica); !canReplicate {
+		if canReplicate, err := replica.CanReplicateFrom(candidateReplica); !canReplicate {
 			// lost due to inability to replicate
 			cannotReplicateReplicas = append(cannotReplicateReplicas, replica)
+			if err != nil {
+				_ = log.Errorf("This replica cannot replicate.\nreplica: %v\nerror: %v", replica, err)
```

Comment:
suggestion: one liner log, same as everywhere else. Instance key is enough, no need to for entire replica struct.
```suggestion
				log.Errorf("chooseCandidateReplica(): error checking CanReplicateFrom(). replica: %v; error: %v", replica.Key, err)
```